src/status_im/ui/screens/chat/sheets.cljs

```diff
@@ -27,40 +26,42 @@
                         :color       colors/gray}}
     (i18n/label helper)]])
 
-(defn chat-actions [{:keys [chat-id contact chat-name]}]
-  [react/view
-   [list-item/list-item
-    {:theme       :action
-     :icon        (multiaccounts/displayed-photo contact)
-     :title       [view-profile {:name   chat-name
-                                 :helper :t/view-profile}]
-     :accessibility-label :view-chat-details-button
-     :accessories [:chevron]
-     :on-press    #(hide-sheet-and-dispatch  [:chat.ui/show-profile chat-id])}]
-   [list-item/list-item
-    {:theme    :action
-     :title    :t/mark-all-read
-     :accessibility-label :mark-all-read-button
-     :icon     :main-icons/check
-     :on-press #(hide-sheet-and-dispatch [:chat.ui/mark-all-read-pressed chat-id])}]
-   [list-item/list-item
-    {:theme    :action
-     :title    :t/clear-history
-     :accessibility-label :clear-history-button
-     :icon     :main-icons/close
-     :on-press #(hide-sheet-and-dispatch [:chat.ui/clear-history-pressed chat-id])}]
-   [list-item/list-item
-    {:theme    :action
-     :title    :t/fetch-history
-     :accessibility-label :fetch-history-button
-     :icon     :main-icons/arrow-down
-     :on-press #(hide-sheet-and-dispatch [:chat.ui/fetch-history-pressed chat-id])}]
-   [list-item/list-item
-    {:theme    :action-destructive
-     :title    :t/delete-chat
-     :accessibility-label :delete-chat-button
-     :icon     :main-icons/delete
-     :on-press #(hide-sheet-and-dispatch [:chat.ui/remove-chat-pressed chat-id])}]])
+(defn one-to-one-chat-actions [{:keys [chat-id]}]
+  (let [photo @(re-frame/subscribe [:chats/photo-path chat-id])
+        contact-name @(re-frame/subscribe [:contacts/contact-name-by-identity chat-id])]
```

Comment:
we know for sure these fields shouldn't change during this view is opened, so instead of creating and disposing of subscriptions we could use direct access to app-db probably in such cases , yeah i think we should consider this idea in general because in that case we will have less "noise" subscriptions, i mean such subscriptions which are run each time we change something but we know for sure result will be the same

Reply:
some stuff is actually already denormalized, for example `chat` now has the `photo` (which never changes), I haven't yet used the denormalized version yet as I am not 100% sure it always has it, but I can change that in a separate PR (we can subscribe only if it's nil for example).